### DISCUSSION TITLE
How to handle open-sourced engines used in numerous games? (BRender, ttengine)

### DISCUSSION THREAD

**Discussion by kparal** [opengaming/osgameclones#1701] [General]
BRender engine was open sourced ([Twitter](https://twitter.com/Foone/status/1521574394761383936)) a few days ago, and it was used in **[many](https://floppy.foone.org/w/BRender#Games_using_the_BRender_Engine)** games. A game engine isn't equal to the whole game (except when it is, like in Doom), and so I'm not sure whether and how to include this in osgameclones:
1. Should it use `type: Official`, even though it's just a sizeable portion of the source code, but not all? (Note that even official releases mostly don't contain non-code assets, so they are also not complete, but this is of course something a bit different).
2. Should it use `type: Tool`, because it is a tool to re-create the game? The current `Tool` description doesn't fully fit, though, it only talks about playing and modding the game, i.e. it sees that from an end-user point of view, not programmer one.
3. Should a new type be created, like `type: Engine`? Or is it perhaps a bit too specific?
4. Should it not be present on osgameclones at all, because it's irrelevant to end users? On the other hand, it's very relevant to programmers, who then create projects relevant to end users, and OSGC add a lot of visibility for OSS projects like these and hopefully attract more programmers to participate. Also, OSGC already includes some `Official` projects like LBA1+2 which might not be even be buildable and they provide no binary releases, so for the end user their value is zero, but they are still present, for programmers.

I personally tend to think that it should be present in OSGC and probably be marked as `Tool` (and `Tool`'s description could be possibly tweaked a bit). What do you think?

A slight complication is that open-sourced BRender is getting [multiple repositories](https://floppy.foone.org/w/BRender#Open_Sourced_BRender) according to its version, which is great from a technical standpoint, but a bit cumbersome for a project like OSGC. Furthermore, it's not even always clear [what game used which engine version](https://floppy.foone.org/w/BRender_versions_used_by_games), and so it might be a bit challenging to decide which game engine repo to link to a particular game (or whether to link to the [personal summary page](https://floppy.foone.org/w/BRender) instead instead of the github repos). But this is a minor problem, I think.

A similar example to BRender is the [Two Tribes engine](https://www.gamingonlinux.com/2021/12/two-tribes-open-sourced-their-game-engine-for-rive-and-toki-tori-2-/), which also covers several games, but again doesn't include all the code for those games.

Update: I created individual tickets for these engines: https://github.com/opengaming/osgameclones/issues/1908 and https://github.com/opengaming/osgameclones/issues/1909.

**Comment by cxong** [comment]
I think it's best to consider this similar to scummvm, which is also another engine that supports many games. We will add all the scummvm games #1818, but like it, we will need to determine which version of brender can run which game before adding it.
